tests/trainer/test_trainer.py

```diff
@@ -548,7 +548,7 @@ def training_step(self, batch, batch_idx):
             return output
 
     model = TestModel()
-    early_stop = EarlyStopping(monitor="loss", patience=0)
+    early_stop = EarlyStopping(monitor="loss", patience=0, check_on_train_epoch_end=True)
```

Comment:
Just want to bring awareness, this changes the behavior for users. 
A user that has monitor on a training metric now needs to set this argument after upgrading, right? Let's make this clear in the changelog, in the "Changed" section?

And are we good to include this in 1.3 during feature freeze here? I assume so since milestone is set to 1.3 but just to make it clear to everyone.

Reply:
Added to the changelog. @edenlightning @tchaton wdyt for 1.3?

Reply:
@awaelchli fwiw this is pretty new behavior: https://github.com/PyTorchLightning/pytorch-lightning/pull/5208

but definitely, it's a change compared to before and i dont think there's a way to make this backward compatible. logging a warning also can get lost. keeping this around longer blocks the general loop refactor we want to do

Reply:
yes, it can't be made compatible. I just want to push for a better changelog. 
Thanks @ananthsub